sourcegen-generator-kotlin/src/main/kotlin/io/micronaut/sourcegen/KotlinPoetSourceGenerator.kt

```diff
@@ -354,8 +329,17 @@ class KotlinPoetSourceGenerator : SourceGenerator {
         enumDef.annotations.stream().map { annotationDef: AnnotationDef -> asAnnotationSpec(annotationDef) }
             .forEach { annotationSpec: AnnotationSpec -> enumBuilder.addAnnotation(annotationSpec) }
 
-        for (enumConstant in enumDef.enumConstants) {
-            enumBuilder.addEnumConstant(enumConstant)
+        enumDef.enumConstants.forEach { (name: String?, exp: ExpressionDef?) ->
+            if (exp != null) {
+                enumBuilder.addEnumConstant(
+                    name,
+                    TypeSpec.anonymousClassBuilder()
+                        .addInitializerBlock(renderExpressionCode(null, MethodDef.builder("").build(), exp))
+                        .build()
+                )
+            } else {
+                enumBuilder.addEnumConstant(name)
+            }
         }
 
         var companionBuilder: TypeSpec.Builder? = null
```

Comment:
Kotlin generator needs to be able to write properties as well, probably.
It would be best to copy the Java tests to make sure that everything works the same.